test/unit_test/eigen_memory_tree_test.cc

```diff
@@ -0,0 +1,148 @@
+// Copyright (c) by respective owners including Yahoo!, Microsoft, and
```

Comment:
New tests related to code in the `core` project should be added to the core test project: https://github.com/VowpalWabbit/vowpal_wabbit/tree/master/vowpalwabbit/core/tests